sw/device/tests/kmac_interrupt_test.c

```diff
@@ -0,0 +1,455 @@
+            "\x19\x85\x0a\x50\x6d\xfa\x9b\x0d\x25\x47\x95\xbe\x78\x57\x86\xc3"
+            "\x19\xa2\x56\x7c\xba\xa5\xe3\x5b\xcf\x8f\xe8\x3d\x94\x3e\x23\xfa"
+            "\x51\x69\xb7\x3a\xdc\x1f\xcf\x8b\x60\x70\x84\xb1\x5e\x6a\x01\x3d"
+            "\xf1\x47\xe4\x62\x56\xe4\xe8\x03\xab\x75\xc1\x10\xf7\x78\x48\x13"
+            "\x6b\xe7\xd8\x06\xe8\xb2\xf8\x68\xc1\x6c\x3a\x90\xc1\x44\x63\x40"
+            "\x70\x38\xcb\x7d\x92\x85\x07\x9e\xf1\x62\xc6\xa4\x5c\xed\xf9\xc9"
+            "\xf0\x66\x37\x5c\x96\x9b\x5f\xcb\xcd\xa3\x7f\x02\xaa\xcf\xf4\xf3"
+            "\x1c\xde\xd3\x76\x75\x70\x88\x54\x26\xbe\xbd\x9e\xca\x87\x7e\x44"
+            "\x67\x4e\x9a\xe2\xf0\xc2\x4c\xdd\x0e\x7e\x1a\xaf\x1f\xf2\xfe\x7f"
+            "\x80\xa1\xc4\xf5\x07\x8e\xb3\x4c\xd4\xf0\x6f\xa9\x4a\x2d\x1e\xab"
+            "\x58\x06\xca\x43\xfd\x0f\x06\xc6\x0b\x63\xd5\x40\x2b\x95\xc7\x0c"
+            "\x21\xea\x65\xa1\x51\xc5\xcf\xaf\x82\x62\xa4\x6b\xe3\xc7\x22\x26"
+            "\x4b",
+        .message_len = 209,
+        .digest = {0x49d25430, 0x03a616f9, 0x3e9c2a9b, 0x8741c1be, 0x8a60a091,
+                   0x366d0e17, 0xe5356048, 0xea3526f9, 0x2a0798ba, 0xb53c3785,
+                   0xf9e32a4e, 0x2b13ce82},
+        .digest_len = DIGEST_LEN_SHA3_384,
+    },
+    {
+        .mode = kDifKmacModeSha3Len512,
+        .message =
+            "\x66\x4e\xf2\xe3\xa7\x05\x9d\xaf\x1c\x58\xca\xf5\x20\x08\xc5\x22"
+            "\x7e\x85\xcd\xcb\x83\xb4\xc5\x94\x57\xf0\x2c\x50\x8d\x4f\x4f\x69"
+            "\xf8\x26\xbd\x82\xc0\xcf\xfc\x5c\xb6\xa9\x7a\xf6\xe5\x61\xc6\xf9"
+            "\x69\x70\x00\x52\x85\xe5\x8f\x21\xef\x65\x11\xd2\x6e\x70\x98\x89"
+            "\xa7\xe5\x13\xc4\x34\xc9\x0a\x3c\xf7\x44\x8f\x0c\xae\xec\x71\x14"
+            "\xc7\x47\xb2\xa0\x75\x8a\x3b\x45\x03\xa7\xcf\x0c\x69\x87\x3e\xd3"
+            "\x1d\x94\xdb\xef\x2b\x7b\x2f\x16\x88\x30\xef\x7d\xa3\x32\x2c\x3d"
+            "\x3e\x10\xca\xfb\x7c\x2c\x33\xc8\x3b\xbf\x4c\x46\xa3\x1d\xa9\x0c"
+            "\xff\x3b\xfd\x4c\xcc\x6e\xd4\xb3\x10\x75\x84\x91\xee\xba\x60\x3a"
+            "\x76",
+        .message_len = 145,
+        .digest = {0xf15f82e5, 0xd570c0a3, 0xe7bb2fa5, 0x444a8511, 0x5f295405,
+                   0x69797afb, 0xd10879a1, 0xbebf6301, 0xa6521d8f, 0x13a0e876,
+                   0x1ca1567b, 0xb4fb0fdf, 0x9f89bc56, 0x4bd127c7, 0x322288d8,
+                   0x4e919d54},
+        .digest_len = DIGEST_LEN_SHA3_512,
+    },
+};
+
+#define DIGEST_LEN_SHAKE_MAX 102
+
+/**
+ * SHAKE test description.
+ */
+typedef struct shake_test {
+  dif_kmac_mode_shake_t mode;
+
+  const char *message;
+  size_t message_len;
+
+  const uint32_t digest[DIGEST_LEN_SHAKE_MAX];
+  size_t digest_len;
+} shake_test_t;
+
+/**
+ * SHAKE tests.
+ */
+const shake_test_t shake_tests[] = {
+    {
+        .mode = kDifKmacModeShakeLen128,
+        .message = "OpenTitan",
+        .message_len = 9,
+        .digest = {0x235a6522, 0x3bd735ac, 0x77832247, 0xc6b12919, 0xfb80eff0,
```

Comment:
Could you please document where you got this digest from? In case of SHA3 tests these were from the `https://csrc.nist.gov`, but as these tests use custom message "OpenTitan", you must have generated the digest yourself somehow, please explain in the `const shake_test_t shake_tests[]` description how the digests were calculated.